install/helm/agones/templates/controller.yaml

```diff
@@ -60,6 +60,10 @@ spec:
         release: {{ .Release.Name }}
         heritage: {{ .Release.Service }}
     spec:
+      {{- if .Values.agones.controller.topologySpreadConstraints }}
+      topologySpreadConstraints:
+        {{- toYaml .Values.agones.controller.topologySpreadConstraints.constraints | nindent 8 }}
```

Comment:
This can be simplified as `{{- toYaml .Values.agones.controller.topologySpreadConstraints| nindent 8 }}`, same for other components.

Reply:
Included it to make the values as an array, otherwise CI will complain:

"e2e-feature-gates": Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(Deployment.spec.template.spec.topologySpreadConstraints): invalid type for io.k8s.api.core.v1.PodSpec.topologySpreadConstraints: got "map", expected "array"

Reply:
Disabling auto merge in case you want to discuss further @gongmax